src/features/dsPreview/state/dsPreviewState.ts

```diff
@@ -21,8 +21,28 @@ export const dsPreviewReducer = createReducer(initialState, {
     state.preview = undefined
     state.loading = true
   },
+  'API_DATASETPREVIEWFETCHDONE_REQUEST': (state) => {
```

Comment:
HELP

This is ugly but again makes things work. JS is weird.

Reply:
hrm, yeah let's deal with the rest & then I'll take over this PR & clean this up